packages/authenticate/lib/login.js

```diff
@@ -1,161 +1,109 @@
+  ws.on('open', async () => {
+    ws.send(JSON.stringify({ type: 'CREATE_USER_API_KEY' }))
+  })
 
-  const data = await body.json()
+  ws.on('message', async (message) => {
+    message = JSON.parse(message.toString())
 
-  if (data.error && data.error === 'pending') {
-    return { state: 'pending', data: { id } }
-  } else if (statusCode === 200) {
-    return { state: 'complete', data }
-  } else {
-    throw new errors.UnableToRetrieveTokensError()
-  }
-}
+    if (message.type === 'CREATE_USER_API_KEY_REQ_ID') {
+      const { reqId } = message.data
+      await opts.onReqId(reqId)
+      return
+    }
 
-async function poll (id, timeout, interval) {
-  const expiresAt = Date.now() + timeout
+    if (message.type === 'CREATE_USER_API_KEY_RESULT') {
+      const { userApiKey } = message.data
+      await opts.onUserApiKey(userApiKey)
 
-  async function check (resolve, reject) {
-    let result
-    try {
-      result = await getTokens(id)
-    } catch (err) {
-      return reject(err)
+      ws.close()
+      return
     }
+    /* c8 ignore next 1 */
+    throw new errors.UnknownMessageTypeError(message.type)
+  })
 
-    const { state, data } = result
-
-    if (Date.now() > expiresAt) {
-      reject(new errors.UserDidNotAuthenticateBeforeExpiryError())
-    } else if (state === 'pending') {
-      setTimeout(check, interval, resolve, reject)
-    } else if (state === 'complete') {
-      resolve(data)
-    /* c8 ignore next 3 */
-    } else {
-      // do nothing, never get here
-    }
-  }
+  ws.on('error', function error (err) {
+    opts.onError(err)
+    ws.close()
+  })
+}
 
-  return new Promise(check)
+function generateVerifyUrl (dashboardHost, reqId) {
+  return `${dashboardHost}/#/?reqId=${reqId}`
 }
 
 export default async function startLogin (_args, print) {
   const args = parseArgs(_args, {
     boolean: 'browser',
-    string: ['claim', 'config']
+    string: ['claim', 'config', 'auth-proxy-host', 'dashboard-host', 'platformatic-home']
   })
 
-  let pltDirPath = path.join(process.env.PLT_HOME, '.platformatic')
+  /* c8 ignore next 2 */
+  const dashboardHost = args['dashboard-host'] || PLT_DASHBOARD_HOST
+  const authProxyHost = args['auth-proxy-host'] || PLT_AUTH_PROXY_HOST
+  const platformaticHome = args['platformatic-home'] || PLT_HOME
+
+  let pltDirPath = path.join(platformaticHome, '.platformatic')
   if (args.config) {
     const stats = await lstat(args.config)
-    if (stats.isDirectory()) throw new errors.ConfigOptionRequiresPathToFileError()
+    if (stats.isDirectory()) {
+      throw new errors.ConfigOptionRequiresPathToFileError()
+    }
     pltDirPath = path.dirname(args.config)
   }
 
-  try {
-    await mkdir(pltDirPath)
-  /* c8 ignore next 2 */
-  } catch {
-  }
+  await mkdir(pltDirPath).catch(() => {})
 
   const config = new ConfigManager({
+    /* c8 ignore next 1 */
     source: args.config || path.join(pltDirPath, 'config.json'),
```

Comment:
Where is this stored by default?

Reply:
`process.env.HOME + '.platformatic' + 'config.json'`